docs/semgrep-appsec-platform/scm-code-access.md

```diff
@@ -0,0 +1,111 @@
+
+#### Required SCM code access scopes
+
+| SCM | Read access scope | Write access scope |
+|----------|-------------|--------------|
+| Azure DevOps | `code:read` | `code:write` |
+| Bitbucket Cloud | `repository:read`<br />`pullrequest:read` | `repository:write`<br />`pullrequest:write` |
+| Bitbucket Data Center | `repository:read` | `repository:write` |
+| GitHub.com and Github Enterprise | `contents:read` | `contents:write` |
+| GitLab and Gitlab Self-Managed | `read_repository` | `write_repository` |
+
+## Grant code access to Semgrep with a private GitHub app
+
+If you already have a private Semgrep GitHub app set up and configured for your deployment that **doesn't** have code access enabled, follow these steps to update the app and grant code access to Semgrep.
+
+:::tip App slug
+If you don't know the name of your app slug, you can find it on the [**Settings > Source code managers** page](https://semgrep.dev/orgs/-/settings/source-code). 
+
+![DESCRIPTION](/img/github-app-slug.png#md-width)
+_**Figure**. Add a description here._
+:::
+
+1. Navigate to the GitHub Application permissions and events page. GitHub Enterprise users must replace the `https://github.com` base URL with the base URL of the GitHub Enterprise instance.
+    1. For organization accounts, go to <code>https://github.com/organizations/<PL>ORGANIZATION_NAME</PL>/settings/apps/<PL>APP_SLUG</PL>/permissions</code>.
+    1. For user accounts, go to <code>https://github.com/settings/apps/<PL>APP_SLUG</PL>/permissions</code>
+2. Expand **Repository Permissions**.
+3. Under **Contents**, change the access level to **Read and write**.
+4. Click **Save Changes**.
+5. At this point, GitHub sends you or your GitHub admin an email to approve the permissions changes. Once approved, Semgrep has code access to your GitHub instance.
+
+## Grant code access to Semgrep with an access token
+
+If you onboarded your repositories using an access token, then you can follow these steps to grant code access to Semgrep.
+
+<Tabs
+    defaultValue="github"
+    values={[
+    {label: 'Azure DevOps Cloud', value: 'ado'},
+    {label: 'Bitbucket Cloud', value: 'bitbucket'},
+    {label: 'Bitbucket Cloud', value: 'bitbucket_datacenter'},
+    {label: 'GitHub', value: 'github'},
+    {label: 'GitLab', value: 'gitlab'},
+    ]}
+>
+
+<TabItem value="ado">
+1. Navigate to the Azure DevDps access token settings page: <code>https://dev.azure.com/<PL>ORGANIZATION_NAME</PL>/_usersSettings/tokens</code>.
+2. Click **New token** to launch the **Create a new personal access token** dialog. Ensure that you assign the `Code: Read` and `Code: Write` scopes to the token, in addition to [any other scopes you may need](/deployment/managed-scanning/azure#prerequisites-and-permissions) for other features you've enabled for your Semgrep deployment.
```

Comment:
I updated this to make it clear that the user needs all existing scopes plus the new ones, otherwise I think people might accidentally break stuff